WindowsStubs/WindowsStubs/TaskUtil.cpp

```diff
@@ -0,0 +1,89 @@
+#include <windows.h>
+#include <cstdio>
+#include <tchar.h>
+#include <iostream>
+#include <string>
+#include <cctype>
+#include <algorithm>
+#include <memory>
+
+
+struct ProcessInfoDeleter {
+    void operator()(PROCESS_INFORMATION* ppi) {
+        if (ppi) {
+            if(ppi->hProcess)
+            CloseHandle(ppi->hProcess);
+            if(ppi->hThread)
+            CloseHandle(ppi->hThread);
+            delete ppi;
+        }
+    }
+};
+
+bool TaskExecuteAndWait(std::string execpath, std::string *pOutput) {
+    STARTUPINFO si;
+    SECURITY_ATTRIBUTES securityAttr;
+    HANDLE inputPipe= INVALID_HANDLE_VALUE, outputPipe= INVALID_HANDLE_VALUE;
+    bool retval = true;
+    std::unique_ptr<PROCESS_INFORMATION,ProcessInfoDeleter> upi(new PROCESS_INFORMATION);
+    PROCESS_INFORMATION *ppi = upi.get();//only for clarity purposes, ownership is not relinquished
+    ZeroMemory(&si, sizeof(si));
+    si.cb = sizeof(si);
+    ZeroMemory(ppi, sizeof(PROCESS_INFORMATION));
+    ZeroMemory(&securityAttr, sizeof(securityAttr));
+    securityAttr.nLength = sizeof(securityAttr);
+    securityAttr.bInheritHandle = TRUE;
+    int len;
+    int slength = (int)execpath.length() + 1;
+    //CP_ACP instructs the API to use the currently set default Windows ANSI codepage
+    len = MultiByteToWideChar(CP_ACP, 0, execpath.c_str(), slength, 0, 0);
+    auto ubuf = std::unique_ptr<wchar_t[]>{ new wchar_t[len] };
+    wchar_t *pbuf = ubuf.get(); //only for clarity
+    MultiByteToWideChar(CP_ACP, 0, execpath.c_str(), slength, pbuf, len);
+    if (pOutput) {
+        if (CreatePipe(&inputPipe, &outputPipe, &securityAttr, 0) != 0) {
+            si.dwFlags = STARTF_USESTDHANDLES;
+            si.hStdInput = NULL;
+            si.hStdOutput = outputPipe;
+            si.hStdError = outputPipe;
+        }
+        else
+            pOutput = nullptr;// set it to null so that later the pipe related functionality is skipped.Allow other functionality to execute.
```

Comment:
Small improvement:
according to the convention: " if one part of an if-else statement uses curly braces, the other part must too".

Example:
// Curly braces around both IF and ELSE required because
// one of the clauses used braces.
if (condition) {
    foo;
} else {
    bar;
}